assimilation_code/modules/assimilation/probit_transform_mod.f90

```diff
@@ -334,38 +349,43 @@ subroutine to_probit_bounded_normal_rh(ens_size, state_ens, p, probit_ens, &
 ! How to handle identical ensemble members is an open question for now. This is also a problem
 ! for ensemble members that are identical to one of the bounds. 
 
+! If the stanard deviation computed for the sample ensemble is not positive, don't know how
+! to do a transform? This can happen when the standard deviation computation is 0 or negative
+! due to computational precision errors. Could try to work around this, but challenge with knowing
+! what to do on the tails where a normal distribution requires a standard deviation.
+
 integer,  intent(in)                          :: ens_size
 real(r8), intent(in)                          :: state_ens(ens_size)
 type(distribution_params_type), intent(inout) :: p
 real(r8), intent(out)                         :: probit_ens(ens_size)
 logical,  intent(in)                          :: use_input_p
 logical,  intent(in)                          :: bounded_below, bounded_above
 real(r8), intent(in)                          :: lower_bound,   upper_bound
+integer,  intent(out)                         :: ierr
 
 ! Probit transform for bounded normal rh.
 integer  :: i
 real(r8) :: quantile(ens_size)
 
-if(use_input_p) then
-   ! Do not know what to do if sd of original ensemble is 0 (or small, work on this later)
-   if(get_bnrh_sd(p) <= 0.0_r8) then
-      ! Just return the original ensemble
-      probit_ens = state_ens 
-      return
-   endif
+! Successful return has ierr 0
+ierr = 0
 
+if(use_input_p) then
+   ! Need this to fail if p isn't available. Is that possible?
    ! Get the quantiles for each of the ensemble members in a BNRH distribution
    call bnrh_cdf_initialized_vector(state_ens, ens_size, p, quantile)
-
 else
    ! Get all the info about the rank histogram cdf
    call bnrh_cdf_params(state_ens, ens_size, bounded_below, bounded_above, &
       lower_bound, upper_bound, p, quantile)
 
-   ! Do not know what to do if sd is 0 (or small, work on this later)
+   ! Fail if sd is not positive (or small, work on this later)
    if(get_bnrh_sd(p) <= 0.0_r8) then
+      ierr = 1
       ! Just return the original ensemble
       probit_ens = state_ens 
```

Comment:
this is quite a nice contract, that if the transform fails, just return the original ensemble.